RecoTracker/MkFitCore/interface/Track.h

```diff
@@ -0,0 +1,800 @@
+
+    static const char* algoint_to_cstr(int algo);
+
+    // ------------------------------------------------------------------------
+
+  protected:
+    TrackState state_;
+    float chi2_ = 0.;
+    float score_ = 0.;
+    short int lastHitIdx_ = -1;
+    short int nFoundHits_ = 0;
+    Status status_;
+    int label_ = -1;
+  };
+
+  //==============================================================================
+  // TrackCand
+  //==============================================================================
+
+  // TrackCand depends on stuff in mkFit/HitStructures, CombCand in particular,
+  // so it is declared / implemented there.
+
+  // class TrackCand : public TrackBase { ... };
+
+  //==============================================================================
+  // Track
+  //==============================================================================
+
+  class Track : public TrackBase {
+  public:
+    Track() {}
+
+    explicit Track(const TrackBase& base) : TrackBase(base) {
+      // Reset hit counters -- caller has to initialize hits.
+      lastHitIdx_ = -1;
+      nFoundHits_ = 0;
+    }
+
+    Track(const TrackState& state, float chi2, int label, int nHits, const HitOnTrack* hits)
+        : TrackBase(state, chi2, label) {
+      reserveHits(nHits);
+      for (int h = 0; h < nHits; ++h) {
+        addHitIdx(hits[h].index, hits[h].layer, 0.0f);
+      }
+    }
+
+    Track(int charge, const SVector3& position, const SVector3& momentum, const SMatrixSym66& errors, float chi2)
+        : TrackBase(charge, position, momentum, errors, chi2) {}
+
+    Track(const Track& t) : TrackBase(t), hitsOnTrk_(t.hitsOnTrk_) {}
+
+    ~Track() {}
+
+    // used for swimming cmssw rec tracks to mkFit position
+    float swimPhiToR(const float x, const float y) const;
+
+    bool canReachRadius(float R) const;
+    float maxReachRadius() const;
+    float zAtR(float R, float* r_reached = nullptr) const;
+    float rAtZ(float Z) const;
+
+    //this function is very inefficient, use only for debug and validation!
+    const HitVec hitsVector(const std::vector<HitVec>& globalHitVec) const {
+      HitVec hitsVec;
+      for (int ihit = 0; ihit < Config::nMaxTrkHits; ++ihit) {
+        const HitOnTrack& hot = hitsOnTrk_[ihit];
+        if (hot.index >= 0) {
+          hitsVec.push_back(globalHitVec[hot.layer][hot.index]);
+        }
+      }
+      return hitsVec;
+    }
+
+    void mcHitIDsVec(const std::vector<HitVec>& globalHitVec,
+                     const MCHitInfoVec& globalMCHitInfo,
+                     std::vector<int>& mcHitIDs) const {
+      for (int ihit = 0; ihit <= lastHitIdx_; ++ihit) {
+        const HitOnTrack& hot = hitsOnTrk_[ihit];
+        if ((hot.index >= 0) && (static_cast<size_t>(hot.index) < globalHitVec[hot.layer].size())) {
+          mcHitIDs.push_back(globalHitVec[hot.layer][hot.index].mcTrackID(globalMCHitInfo));
+          //globalMCHitInfo[globalHitVec[hot.layer][hot.index].mcHitID()].mcTrackID());
```

Comment:
remove

Reply:
Addressed in Round 2 follow-up PR trackreco/cmssw#76 commit https://github.com/trackreco/cmssw/pull/76/commits/6f728696277e15a8502cb9e080061ad7d504f604